orangecontrib/text/widgets/owpreprocess.py

```diff
@@ -138,7 +138,12 @@ class SpinBox(QSpinBox):
         def validate(self, *args):
             # accept empty input
             valid, text, pos = super().validate(*args)
-            return 2 if valid else 0, text, pos
+            new_state = (
+                QValidator.State.Acceptable
+                if valid == QValidator.State.Acceptable
+                else QValidator.State.Invalid
+            )
+            return new_state, text, pos
 
         def valueFromText(self, text: str) -> int:
             return max(int(text) if text else self.minimum(), self.minimum())
```

Comment:
When writing this, the intended behaviour was, that the user could delete everything in the upper limit spinbox (or write some smaller value), and it would default to it's minimum (interpreted as no upper bound). Now it won't allow for that. I feel like the behaviour has changed with Qt6. But I could have just written it poorly in the first place. At any rate, I would either delete both these functions completely, if the current behaviour is okay, or find a way to allow for the aforementioned behaviour.

Reply:
@noahnovsak, thank you for noticing this one. It was my mistake. I wrongly changed the condition `2 if valid else 0` to `QValidator.State.Acceptable if valid == QValidator.State.Acceptable else QValidator.State.Invalid`, which is indeed not the same. I think now it should work.